tiny_cnn/layers/convolutional_layer.h

```diff
@@ -489,15 +493,21 @@ class convolutional_layer : public layer<Activation> {
         return img;
     }
 
+    virtual void set_worker_count(cnn_size_t worker_count) {
+        Base::set_worker_count(worker_count);
+        conv_layer_worker_storage_.resize(worker_count);
+        init();
+    }
+
 private:
     void init() {
-        for (cnn_size_t i = 0; i < CNN_TASK_SIZE; i++) {
+        for (conv_layer_worker_specific_storage& cws : conv_layer_worker_storage_) {
             if (pad_type_ == padding::same) {
-                prev_out_buf_[i] = new vec_t(in_padded_.size(), float_t(0));
```

Comment:
I wonder if there was a memory leak here. Looks like the allocated `prev_out_buf_` was never `free`d?

Anyway fixed now by avoiding manual memory management.